components/brave_vpn/browser/connection/ikev2/win/utils.cc

```diff
@@ -84,6 +86,25 @@ absl::optional<std::string> SetCredentials(LPCTSTR phone_book_path,
   return absl::nullopt;
 }
 
+std::wstring TryToCreateEmptyPhoneBookFile() {
+  base::FilePath dir;
+  if (!base::PathService::Get(base::DIR_ROAMING_APP_DATA, &dir)) {
+    return L"";
+  }
+
+  dir = dir.Append(L"Microsoft")
+            .Append(L"Network")
+            .Append(L"Connections")
+            .Append(L"Pbk");
+  if (!base::CreateDirectoryAndGetError(dir, nullptr)) {
+    return L"";
+  }
+
+  base::FilePath phone_book_path = dir.Append(L"rasphone.pbk");
+  base::ScopedFILE file_stream(base::OpenFile(phone_book_path, "w"));
+  return file_stream ? phone_book_path.value() : L"";
```

Comment:
Seems like it is enough just return the path and the system creates file automatically. I just checked on my side and pbk file was created inside `AppData\Roaming\Microsoft\Network\Connections\Pbk` by `RasSetEntryProperties` call from browser launched as user.

Reply:
@spylogsster you mean basically always return the full path? (even if the file doesn't exist?)

Reply:
I think it could help to detect error like permission in our code before going to ras call.